src/ring_theory/unique_factorization_domain.lean

```diff
@@ -361,10 +361,52 @@ unique_factorization_monoid.of_exists_prime_factors (by
   { convert eif,
     simp_rw irreducible_iff_prime_of_exists_unique_irreducible_factors eif uif })
 
+namespace unique_factorization_monoid
+variables [comm_cancel_monoid_with_zero α] [decidable_eq α]
+variables [unique_factorization_monoid α]
+/-- Noncomputably determines the multiset of prime factors. -/
+noncomputable def factors'' (a : α) : multiset α := if h : a = 0 then 0 else
```

Comment:
I'm not so fond of the `''`. Can we choose a name(space)` that avoids this? Dare we place it in the root namespace?

(Also, `factors'` is in the `associates` namespace, so you could at least get rid of 1 `'`.

I think my preference would be to call this `factors` and to rename the other one to `normalized_factors`.

Reply:
That makes sense, I'll take a look how much work it is :)